src/Dhall/Parser.hs

```diff
@@ -61,710 +62,9 @@ import qualified Text.Parser.Combinators
-
-_or :: Parser ()
-_or = reserved "||"
-
-_plus :: Parser ()
-_plus = reserved "+"
-
-_textAppend :: Parser ()
-_textAppend = reserved "++"
-
-_listAppend :: Parser ()
-_listAppend = reserved "#"
-
-_and :: Parser ()
-_and = reserved "&&"
-
-_times :: Parser ()
-_times = reserved "*"
-
-_doubleEqual :: Parser ()
-_doubleEqual = reserved "=="
-
-_notEqual :: Parser ()
-_notEqual = reserved "!="
-
-_dot :: Parser ()
-_dot = reserved "."
-
-_openBrace :: Parser ()
-_openBrace = reserved "{"
-
-_closeBrace :: Parser ()
-_closeBrace = reserved "}"
-
-_openBracket :: Parser ()
-_openBracket = reserved "["
-
-_closeBracket :: Parser ()
-_closeBracket = reserved "]"
-
-_openAngle :: Parser ()
-_openAngle = reserved "<"
-
-_closeAngle :: Parser ()
-_closeAngle = reserved ">"
-
-_bar :: Parser ()
-_bar = reserved "|"
-
-_comma :: Parser ()
-_comma = reserved ","
-
-_openParens :: Parser ()
-_openParens = reserved "("
-
-_closeParens :: Parser ()
-_closeParens = reserved ")"
-
-_colon :: Parser ()
-_colon = reserved ":"
-
-_at :: Parser ()
-_at = reserved "@"
-
-_combine :: Parser ()
-_combine = do
-    void (Text.Parser.Char.char '∧' <?> "\"∧\"") <|> void (Text.Parser.Char.text "/\\")
-    whitespace
-
-_combineTypes :: Parser ()
-_combineTypes = do
-    void (Text.Parser.Char.char '⩓' <?> "\"⩓\"") <|> void (Text.Parser.Char.text "//\\\\")
-    whitespace
-
-_prefer :: Parser ()
-_prefer = do
-    void (Text.Parser.Char.char '⫽' <?> "\"⫽\"") <|> void (Text.Parser.Char.text "//")
-    whitespace
-
-_lambda :: Parser ()
-_lambda = do
-    _ <- Text.Parser.Char.satisfy predicate
-    whitespace
-  where
-    predicate 'λ'  = True
-    predicate '\\' = True
-    predicate _    = False
-
-_forall :: Parser ()
-_forall = do
-    void (Text.Parser.Char.char '∀' <?> "\"∀\"") <|> void (Text.Parser.Char.text "forall")
-    whitespace
-
-_arrow :: Parser ()
-_arrow = do
-    void (Text.Parser.Char.char '→' <?> "\"→\"") <|> void (Text.Parser.Char.text "->")
-    whitespace
-
-doubleLiteral :: Parser Scientific
-doubleLiteral = (do
-    sign <-  fmap (\_ -> negate) (Text.Parser.Char.char '-')
-         <|> pure id
-    a    <-  Text.Parser.Token.scientific
-    return (sign a) ) <?> "double literal"
-
-integerLiteral :: Parser Integer
-integerLiteral = (do
-    let positive = fmap (\_ -> id    ) (Text.Parser.Char.char '+')
-    let negative = fmap (\_ -> negate) (Text.Parser.Char.char '-')
-    sign <- positive <|> negative
-    a <- Text.Parser.Token.natural
-    return (sign a) ) <?> "integer literal"
-
-naturalLiteral :: Parser Natural
-naturalLiteral = (do
-    a <- Text.Parser.Token.natural
-    return (fromIntegral a) ) <?> "natural literal"
-
-identifier :: Parser Var
-identifier = do
-    x <- label
-
-    let indexed = do
-            _ <- Text.Parser.Char.char '@'
-            Text.Parser.Token.natural
-
-    n <- indexed <|> pure 0
-    return (V x n)
+import Dhall.Parser.Combinators
+import Dhall.Parser.Token
+import Dhall.Parser.Expression
```

Comment:
I think you need to add `Dhall.Parser.Expression` to version control because I don't see it in the set of changes

Reply:
Oh yes, sorry, I've pushed it now (and also added the modules to cabal file).

Thanks for looking at it, it's still a work in progress, so not really ready for final review. Let's discuss it tomorrow.